### ISSUE TITLE
Blocks: load payment request button configuration in the Blocks Checkout PHP handler instead of relying on configuration loaded for the Shortcode checkout

### ISSUE THREAD

**Issue by reykjalin** [Automattic/woocommerce-payments#2604]
Similar to the changes introduced in Stripe: https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1585

## Problem

We currently rely on the JS configuration loaded for the shortcode checkout [here](https://github.com/Automattic/woocommerce-payments/blob/b4e96317ce9e91ca1dbf49b810e6c46d7c9118b9/includes/class-wc-payments-payment-request-button-handler.php#L668-L704) in the block-based checkout (see for example how we use the `wcpayPaymentRequestParams` global [here](https://github.com/Automattic/woocommerce-payments/blob/b4e96317ce9e91ca1dbf49b810e6c46d7c9118b9/client/payment-request/blocks/index.js#L23-L25)).

This is bad for **at least 2 reasons**:

* We already have specific handlers for the Payment Request Buttons in the block-based integration JS, and the shortcode JS may contain some duplicate handlers that can potentially mess with the payment flow.
* Loading unnecessary JS has performance impacts that can significantly slow down the checkout experience for no gain.

## Proposed solution

1. Extract the configuration object into it's own function (similar to what we did [in the Stripe plugin](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1585/files#diff-e2b9a082adab276b037cd70074b84f278c903bfed3c4b3d77a46aa04bcbfb78aR669-R676)).
2. Use that function to load the payment request configuration into the Blocks settings [here](https://github.com/Automattic/woocommerce-payments/blob/a10fe849a19a07a2d2b865e2c0363fcbb78a75c2/includes/class-wc-payments-blocks-payment-method.php#L62-L71).
3. Stop using the `wcpayPaymentRequestParams` global in the block integration JS.

Look at https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1585 for inspiration.

## Things to consider and test

1. Are we relying on some JS from the shortcode JS in the block integration JS?
    * If yes, then we should remove that dependency and implement everything in the block integration JS.
2. How will this affect other theme builders with their own custom cart and checkout blocks, if at all?
    * Themes to look out for here that we've made mistakes with in the past (see pc4etw-aT-p2): CartFlows, possibly others.
3. What possible side effects could a refactor like this have?
4. Can we introduce unit (or even e2e) tests here that help us make sure the refactor doesn't break any existing functionality? Do these tests already exist?

**Comment by vbelolapotkov** [comment]
@c-shultz this one is related to checkout, but doesn't show up in your board, could you please check?

**Comment by frosso** [comment]
The Payment Request Button subsystem this issue is built around doesn't exist anymore - it was superseded by the Express Checkout (ECE) rework in #11144 and #11182. `includes/class-wc-payments-payment-request-button-handler.php`, the `client/payment-request/` directory, and the `wcpayPaymentRequestParams` global are all gone.

More importantly, the actual refactor requested here is effectively done. ECE no longer has Blocks piggyback on the shortcode's loaded config: `class-wc-payments-express-checkout-button-handler.php` localizes `wcpayExpressCheckoutParams` separately to the dedicated ECE script and the Blocks script, and `register_ece_data_for_block_editor()` registers config via the WooCommerce Blocks AssetDataRegistry (`ece_data`) for the block editor. Block vs shortcode button JS is also split into separate directories (`client/express-checkout/block-buttons/` and `.../shortcode-buttons-express/`), which addresses the duplicate-handler and unnecessary-JS concerns.

Closing as obsolete - the subsystem was rewritten and the requested change has been realized in the new architecture.
